### PR TITLE
feat(cli): support configuring the test tool in the config file

### LINKED ISSUE #14263: Consider adding config for `deno test` to deno.jsonc
Wanted to encode my `--ignore` paths into a config file yesterday when I noticed that configuring `deno test` in the configuration file is not supported yet!

Would be great to have a new top-level `"test"` config object alongside `"lint"` and `"fmt"`.

Thanks!

### PR DESCRIPTION
**Pull request by roj1512**
This allows the user to use the config file to configure the included and excluded patterns of the test subcommand.

It has many use cases, one of them is for example when the project uses dnt to transform the tests to work on Node.js, and place them in a directory called `out/`. The test tool should not look for test files in that directory, and instead of providing the CLI args to exclude that directory everytime, the user can just have a config file like the following:

```jsonc
{
  "test": {
    "files": {
      "exclude": ["./out/"]
    }
  }
}
```

This is just a single example, there might be more examples that will define the usefulness of the existence of this feature.

- Closes #14263

### CODE REVIEW

**Comment by dsherret** [comment]
Related: #14263

File: cli/lsp/language_server.rs [hunk 1]
```diff
@@ -121,6 +122,8 @@ pub struct Inner {
   maybe_import_map_uri: Option<Url>,
   /// An optional configuration for linter which has been taken from specified config file.
   pub maybe_lint_config: Option<LintConfig>,
+  /// An optional configuration for the test subcommand which has been taken from specified config file.
+  pub maybe_test_config: Option<TestConfig>,
```
Comment (dsherret) [inline cli/lsp/language_server.rs:126]: Is this used anywhere? I don't think we don't need to bother with this for now and could just revert this file?

Comment (roj1512) [reply cli/lsp/language_server.rs]: Done.

**Comment by bartlomieju** [comment]
We've discussed this PR at the last CLI design meeting and we're in agreement this is useful.

**Review by dsherret — APPROVED** [review]
LGTM. Thanks @roj1512!

File: cli/tests/testdata/test/collect/deno.jsonc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "test": {
+    "files": {
+      "exclude": ["./collect/ignore"]
```
Comment (dsherret) [inline cli/tests/testdata/test/collect/deno.jsonc:4]: One more test like this, but with "include" and this should be good to merge.

Comment (roj1512) [reply cli/tests/testdata/test/collect/deno.jsonc:4]: What if I add the include to this current one?

Comment (dsherret) [reply cli/tests/testdata/test/collect/deno.jsonc:4]: We just need a test that demonstrates they both work. I think there's only two files in this `collect` directory so adding an include might not demonstrate it works.

**Comment by roj1512** [comment]
@dsherret I added the include test. The last failed CI is not related to my changes.

**Review by dsherret — APPROVED** [review]
LGTM. Thanks @roj1512—this will be really nice to have.

**Pull request merged**